libs/telemetry/mavlink/CMakeLists.txt

```diff
@@ -19,10 +19,16 @@ cmake_minimum_required(VERSION 3.20.0)
 
 # Generate message headers
 set(XML_FILES
-    ${CMAKE_SOURCE_DIR}/tools/mavlink/message_definitions/v1.0/common.xml
+    ${CMAKE_SOURCE_DIR}/../tools/mavlink/message_definitions/v1.0/common.xml
 )
 set(GENERATED_DIR ${CMAKE_BINARY_DIR}/mavlink/generated)
-set(GENERATOR_SCRIPT ${CMAKE_SOURCE_DIR}/tools/mavlink/pymavlink/tools/mavgen.py)
+set(GENERATOR_SCRIPT ${CMAKE_SOURCE_DIR}/../tools/mavlink/pymavlink/tools/mavgen.py)
```

Comment:
There is a lot of `${CMAKE_SOURCE_DIR}/../` use, perhaps we can create a new variable for that?